src/synthesizer/particle/stars.py

```diff
@@ -612,6 +594,11 @@ def _aperture_mask(self, aperture_radius):
                 Radius of spherical aperture in kpc
         """
 
+        if self.centre is None:
+            raise ValueError(
+                "Centre of star particles must be set to use aperture mask."
+            )
+
         distance = np.sqrt(
```

Comment:
There's a property attached to all particles called something like `centered_coordinates` which automatically hands the centered coordinates and also ensures centre has been set at the same time. 

You could remove the check by using that property here.

Reply:
moved to particle parent class, and simplified to use existing methods